pkg/router/router_test.go

```diff
@@ -228,6 +365,32 @@ func TestConfigTemplateExecution(t *testing.T) {
 			t.Fatalf("Template execution failed: %v", e)
 		}
 	}
+
+	// check the generated config
+	config := path.Join(h.workdir, "conf", "haproxy.config")
+	contents, err := ioutil.ReadFile(config)
+	if err != nil {
+		t.Fatalf("Failed to read the generated config: %v", err)
+	}
+
+	for name, expectations := range tests {
+		for _, expectation := range expectations {
+			createExpectation, ok := expectation.(mustCreateWithConfig)
+			if !ok {
+				continue
```

Comment:
Could we call `t.Fatalf` here to avoid silently skipping a test?  Alternatively, could we use a `map[string]mustCreateWithConfig` type for the test cases?

Reply:
Test cases' type changed to `map[string][]mustCreateWithConfig` to avoid the cast.